ir/src/constraints/graph.rs

```diff
@@ -485,6 +497,42 @@ impl AlgebraicGraph {
             ))),
         }
     }
+
+    /// Inserts a list folding expression into the graph and returns the resulting expression
+    /// details.
+    ///
+    /// # Errors
+    /// Panics if the list is empty.
+    /// Returns an error if the list cannot be unfolded properly.
```

Comment:
should we give these bullet points? I think we usually do when there are multiple error cases, right?